File: easy/merge.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
  let num = m+n-1;
  let i=m-1;
  let j=n-1;
  while(i>=0||j>=0) {
    if(i<0){
      nums1[num]=nums2[j];
      num--;
      j--;
    }else if(j<0){
      nums1[num]=nums1[i];
      num--;
      i--;
    }else if(nums1[i]<nums2[j]) {
  nums1[num]=nums2[j];
  num--;
  j--;
}else{
  nums1[num]=nums1[i];
  num--;
  i--;
}
}
return nums1;
};
let nums1=[1,2,3,0,0,0];
let nums2=[2,5,6];
let m=3;
let n=3;
console.log(merge(nums1,m,nums2,n));

File: easy/merge.cpp
// class Solution {
// public:
//     void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
//         int i=m-1;
//         int j=n-1;
//         int p=m+n-1;
//         int q;
//         while(i>=0||j>=0)
//         {
//             if(i<0){
//                 q=nums2[j--];
//             }else if(j<0){
//                 q=nums1[i--];
//             }else
//             if(nums1[i]>nums2[j]){
//                 q=nums1[i--];
//             }else {
//                 q=nums2[j--];
//             }
//             nums1[p--]=q;
//         }
//     }
// };


// 使用了尾部插入方式
// 将nums1和nums2的最后一位的元素进行对比，将大的放在nums1的尾部，如果nums1的尾部更大，则nums2尾部的指针不变，
// 将nums1尾部的指针往前移一位，否则将nums2的值赋给nums1的尾部，同时将nums2的指针向前移一位
// 然后重复操作就能得到结果